packages/components/src/mobile/hr/index.native.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { View, Text } from 'react-native';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+
+function HRLine( { color, lineStyle } ) {
+	const lineStyles = [
+		styles.hr__line,
+		lineStyle,
+		color && { backgroundColor: color },
+	];
+	return <View style={ lineStyles } />;
+}
+
+function HRText( { text, textStyle } ) {
+	const textStyles = [ styles.hr__text, textStyle ];
+	return <Text style={ textStyles }>{ text }</Text>;
+}
+
+function HR( {
+	color,
+	lineStyle,
+	marginLeft = 8,
+	marginRight = 8,
+	text,
+	textStyle,
```

Comment:
Related to my comments regarding composition, I wonder if these custom props could be simplified and/or avoided if we relied upon composition. We could instead rely upon the de facto `style` prop WDYT? 

```js
<HorizontalRule style={{ marginLeft: 16, marginRight: 16 }}>
  <HorizontalRule.Text style={{ color: "green" }}>
    Read More
  </HorizontalRule.Text>
</HorizontalRule>;
```

I believe we could reduce the surface area of the component API: 

* `HorizontalRule`: `style` and `children`
* `HorizontalRule.Text`: `style`